### PR TITLE
Tests for imageGallery

### PR DESCRIPTION
**Pull request by adityanaag3**
### What does this PR do?
Adds tests for imageGallery component

## The PR fulfills these requirements:

[x] Tests for the proposed changes have been added/updated.
[x] Code linting and formatting was performed.

### CODE REVIEW

File: es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        // Create element
+        const element = createElement('c-image-gallery', {
+            is: ImageGallery
+        });
+
+        element.items = mockSpacesList;
+        document.body.appendChild(element);
+
+        return Promise.resolve().then(() => {
+            const lightningLayoutEl = element.shadowRoot.querySelector(
```
Comment (pozil) [inline es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js]: Drop that layout check, we don't need to test that static part of the layout. We need only to check the image tile.

File: es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                'lightning-layout'
+            );
+            expect(lightningLayoutEl).not.toBeNull();
+
+            const imageTileElements = element.shadowRoot.querySelectorAll(
+                'c-image-tile'
+            );
+            expect(imageTileElements.length).toBe(mockSpacesList.length);
```
Comment (pozil) [inline es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js:28]: We also need to check that we are displaying the right image. Test at least the first `image-tile` item and a few attributes (no need to be exhaustive).

File: es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        });
+    });
+
+    it('click on an image tile', () => {
```
Comment (pozil) [inline es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js]: Test name needs to be more descriptive: what's the expected behavior?

File: es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        document.body.appendChild(element);
+
+        // Mock handler for child event
+        const handler = jest.fn();
+        // Add event listener to catch child event
+        element.addEventListener('itemselect', handler);
+
+        return Promise.resolve()
+            .then(() => {
+                const lightningLayoutEl = element.shadowRoot.querySelector(
```
Comment (pozil) [inline es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js]: Drop layout check

File: es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+                const imageTileElement = element.shadowRoot.querySelector(
+                    'c-image-tile'
+                );
+
+                imageTileElement.click();
+                expect(element.items[0].selected).toBe(true);
+            })
+            .then(() => {
+                expect(handler).toHaveBeenCalled();
+                expect(handler.mock.calls[0][0].detail).toEqual(
```
Comment (pozil) [inline es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js:86]: I suggest to use the more fluent alias for this: `toHaveBeenCalledWith()`

File: es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+                    EVENT_DETAIL_PARAMETER
+                );
+
+                const imageTileElement = element.shadowRoot.querySelector(
```
Comment (pozil) [inline es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js]: You're testing too many things here. This needs to go in a separate unselect test.

**Review by pozil — CHANGES_REQUESTED** [review]

File: es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js (hunk 1 above)
Comment (adityanaag3) [reply es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js]: Done

File: es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js (hunk 2 above)
Comment (adityanaag3) [reply es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js:28]: Done. Checking for name and Picture URL, which are the only 2 fields being shown.

File: es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js (hunk 3 above)
Comment (adityanaag3) [reply es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js]: Done. Split it into event fire check and selection toggle check.

File: es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js (hunk 4 above)
Comment (adityanaag3) [reply es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js]: Done

File: es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js (hunk 5 above)
Comment (adityanaag3) [reply es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js:86]: As discussed, toHaveBeenCalledWith() is not doing a deep compare.

File: es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js (hunk 6 above)
Comment (adityanaag3) [reply es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js]: Done.

File: es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        element.items = mockSpacesList;
+        document.body.appendChild(element);
+
+        return Promise.resolve().then(() => {
+            const imageTileElement = element.shadowRoot.querySelector(
+                'c-image-tile'
+            );
+
+            imageTileElement.click();
+            expect(element.items[0].selected).toBe(true);
```
Comment (pozil) [inline es-base-styles/main/default/lwc/imageGallery/__tests__/imageGallery.test.js]: Not a blocker but you could use `isTruthy()` or `isFalsy()` in those checks.

**Review by pozil — APPROVED** [review]

**Pull request merged**